.github/workflows/golang-ci-test.yml

```diff
@@ -0,0 +1,35 @@
+name: Test Meshery-App Mesh Workflow
+on:
+  push:
+    branches:
+      - "*"
+    tags:
+      - "v*"
+  pull_request:
+    branches:
+      - master
+jobs:
+  tests:
+     name: Tests
+     env:
+       ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+     runs-on: ubuntu-latest
+     steps:
+     - name: Check out code
+       uses: actions/checkout@main
+       with:
+        fetch-depth: 1
+     - name: Setup Go
+       uses: actions/setup-go@main
+       with:
+        go-version: 1.19
+        check-latest: true
+        cache: true
+     - name: Create cluster using KinD
```

Comment:
@acald-creator Why does running the unit tests require a kind cluster?

Reply:
Also this is reflected in the Makefile as well

```bash
## Run Golang tests
test: dep-check
	export CURRENTCONTEXT="$(kubectl config current-context)" 
	echo "current-context:" ${CURRENTCONTEXT} 
	export KUBECONFIG="${HOME}/.kube/config"
	echo "environment-kubeconfig:" ${KUBECONFIG}
	GOPROXY=direct GOSUMDB=off GO111MODULE=on go test -v ./...
```